neo/core/analogsignal.py

```diff
@@ -448,7 +448,7 @@ def _pp(line):
     def time_index(self, t):
         """Return the array index (or indices) corresponding to the time (or times) `t`"""
         i = (t - self.t_start) * self.sampling_rate
-        i = np.rint(i.simplified.magnitude).astype(np.int)
+        i = np.rint(i.simplified.magnitude).astype(np.int64)
```

Comment:
@samuelgarcia in case just `int` is provided as type here, wouldn't this automatically pick the suitable integer length that fits the data?

Reply:
Handling time with int32 is super dangerous.
I think forcing to int64 is a good choice.

Reply:
Ok, then let's go for int64 for AnalogSignals times for now. If this becomes a memory issue in the future we might reconsider this choice.